src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -1400,138 +1400,150 @@ void TaskSketcherElements::updateVisibility()
+                // by element number.
+                for (int i = 0; i < countItems; i++) {
+                    ElementItem* item = static_cast<ElementItem*>(ui->listWidgetElements->item(i));
+                    if (item->ElementNbr == ElementId) {
+                        item->isLineSelected = select;
+                        modified_item = item;
+                        SketcherGui::scrollTo(ui->listWidgetElements, i, select);
+                        break;
                     }
                 }
-                else if (shapetype.size() > 6 && shapetype.substr(0, 6) == "Vertex") {
-                    QRegularExpression rx(QStringLiteral("^Vertex(\\d+)$"));
-                    QRegularExpressionMatch match;
-                    boost::ignore_unused(expr.indexOf(rx, 0, &match));
-                    if (match.hasMatch()) {
-                        bool ok;
-                        int ElementId = match.captured(1).toInt(&ok) - 1;
-                        if (ok) {
-                            // Get the GeoID&Pos
-                            int GeoId;
-                            Sketcher::PointPos PosId;
-                            sketchView->getSketchObject()->getGeoVertexIndex(
-                                ElementId, GeoId, PosId);
-
-                            int countItems = ui->listWidgetElements->count();
-                            for (int i = 0; i < countItems; i++) {
-                                ElementItem* item =
-                                    static_cast<ElementItem*>(ui->listWidgetElements->item(i));
-                                if (item->ElementNbr == GeoId) {
-                                    modified_item = item;
-                                    switch (PosId) {
-                                        case Sketcher::PointPos::start:
-                                            item->isStartingPointSelected = select;
-                                            break;
-                                        case Sketcher::PointPos::end:
-                                            item->isEndPointSelected = select;
-                                            break;
-                                        case Sketcher::PointPos::mid:
-                                            item->isMidPointSelected = select;
-                                            break;
-                                        default:
-                                            break;
-                                    }
-                                    break;
-                                }
-                            }
-                        }
+            }
+            else if (boost::starts_with(shapetype, "ExternalEdge")) {
+                QRegularExpression rx(QStringLiteral("^ExternalEdge(\\d+)$"));
+                QRegularExpressionMatch match;
+                boost::ignore_unused(expr.indexOf(rx, 0, &match));

```

Comment:
Same here, you could try:
```suggestion
                expr.indexOf(rx, 0, &match);
```